packages/inter-protocol/src/auction/auctioneer.js

```diff
@@ -0,0 +1,284 @@
+
+        const collShare = makeRatioFromAmounts(collatRaise, totCollDeposited);
+        const currShare = makeRatioFromAmounts(currencyRaise, totCollDeposited);
+        /** @type {import('@agoric/zoe/src/contractSupport/atomicTransfer.js').TransferPart[]} */
+        const transfers = [];
+        let currencyLeft = currencyRaise;
+        let collateralLeft = collatRaise;
+
+        // each depositor gets as share that equals their amount deposited
+        // divided by the total deposited multplied by the currency and
+        // collateral being distributed.
+        for (const { seat, amount } of deposits.get(brand).values()) {
+          const currPortion = floorMultiplyBy(amount, currShare);
+          currencyLeft = AmountMath.subtract(currencyLeft, currPortion);
+          const collPortion = floorMultiplyBy(amount, collShare);
+          collateralLeft = AmountMath.subtract(collateralLeft, collPortion);
+          transfers.push([currencySeat, seat, { Currency: currPortion }]);
+          transfers.push([collateralSeat, seat, { Collateral: collPortion }]);
+        }
+
+        // TODO The leftovers should go to the reserve, and should be visible.
+        const keyword = brandToKeyword.get(brand);
+        transfers.push([
+          currencySeat,
+          reserveFunds,
+          { Currency: currencyLeft },
+        ]);
+        transfers.push([
+          collateralSeat,
+          reserveFunds,
+          { Collateral: collateralLeft },
+          { [keyword]: collateralLeft },
+        ]);
+        atomicRearrange(zcf, harden(transfers));
+
+        for (const { seat } of depositsForBrand) {
+          seat.exit();
+        }
+      }
+    }
+  };
+
+  const { publicMixin, creatorMixin, makeFarGovernorFacet, params } =
+    await handleParamGovernance(
+      zcf,
+      privateArgs.initialPoserInvitation,
+      // @ts-expect-error XXX How to type this?
+      auctioneerParamTypes,
+      privateArgs.storageNode,
+      privateArgs.marshaller,
+    );
+
+  const tradeEveryBook = () => {
+    const bidScalingRatio = makeRatio(
+      currentDiscountRateBP,
+      brands.Currency,
+      BASIS_POINTS,
+    );
+
+    [...books.entries()].forEach(([_collateralBrand, book]) => {
+      book.settleAtNewRate(bidScalingRatio);
+    });
+  };
+
+  const driver = Far('Auctioneer', {
+    reducePriceAndTrade: () => {
+      trace('reducePriceAndTrade');
+
+      natSafeMath.isGTE(currentDiscountRateBP, params.getDiscountStep()) ||
+        Fail`rates must fall ${currentDiscountRateBP}`;
+
+      currentDiscountRateBP = natSafeMath.subtract(
+        currentDiscountRateBP,
+        params.getDiscountStep(),
+      );
+
+      tradeEveryBook();
+    },
+    finalize: () => {
+      trace('finalize');
+      distributeProceeds();
+    },
+    startRound() {
+      trace('startRound');
+
+      currentDiscountRateBP = params.getStartingRate();
+      [...books.entries()].forEach(([_collateralBrand, book]) => {
```

Comment:
please,
```suggestion
      for (const book of Object.values(books)) {
```

Reply:
`for (const book of Object.values(books))` gives me a `book` that doesn't make any sense.

`for (const book of books.values()) {` works much better. 

Is that surprising to you?

Reply:
my error reading `books` as an object instead of a Map. `books.values()` is unsurprisingly correct :)